rust/rust.bzl

```diff
@@ -629,15 +663,14 @@ _rust_toolchain_attrs = {
 
 _rust_library_attrs = _rust_common_attrs + {
     "crate_type": attr.string(),
+    "platform": attr.string(),
 }
 
 rust_library = rule(
     _rust_library_impl,
     attrs = _rust_library_attrs + _rust_toolchain_attrs,
     fragments = ["cpp"],
-    outputs = {
-        "rust_lib": "lib%{name}.rlib",
-    },
+    outputs = _get_crate_type_and_target_outputs,
```

Comment:
This does not seem idiomatic Skylark. Also, it is unclear to me how this would actually work where we pass in a function for the `outputs` parameter of `rule`. Is this an obscure Bazel/Skylark feature? I do not see it documented as a way to specify outputs in both the [Skylark documentation](https://bazel.build/versions/master/docs/skylark/rules.html#output-files) and the [Skylark Cookbook](https://bazel.build/versions/master/docs/skylark/cookbook.html#outputs-default).

@laurentlb @dslomov Can you comment on this?